<launch>
    
    <!-- Default cameras to first two found. This will most likely be fine in
         most cases. If this is not desired, or other config is required, use
         rosrun pointgrey_camera_driver list_cameras. -->
    <arg name="serial" default="0" />
    <arg name="right_cam_serial" default="1" />

    <arg name="calibrated" default="0" />


    <group ns="stereo_vision">

        <node pkg="nodelet" type="nodelet" name="stereo_nodelet_manager" 
              args="manager" />
        
        <node pkg="nodelet" type="nodelet" name="stereo_nodelet"
              args="load pointgrey_camera_driver/PointGreyStereoCameraNodelet 
                    stereo_nodelet_manager" >
            <param name="frame_id" value="stereo" />
            <param name="frame_rate" value="15" />
            <param name="first_namespace" value="left_cam" />
            <param name="second_namespace" value="right_cam" />
            <param name="format7_color_coding" value="raw16" />
            <param name="serial" value="$(arg serial)"/>
            

            <!-- Use camera_calibration package to create files -->
            <param name="camera_info_url" if="$(arg calibrated)"
                value="file://$(env HOME)/.ros/camera_info/$(arg serial).yaml"/>
            <param name="second_info_url" if="$(arg calibrated)"
                value="file://$(env HOME)/.ros/camera_info/$(arg right_cam_serial.yaml" />
            
        </node>

        <!-- Nodelet for the left camera -->
        <node pkg="nodelet" type="nodelet" name="image_proc_debayer_left"
            args="load image_proc/debayer stereo_nodelet_manager">
            <remap from="image_raw" to="left/image_raw" />
            <remap from="image_mono" to="left/image_mono" />
            <remap from="image_color" to="left/image_color" />
        </node>

        <!-- Nodelet for the right camera -->
        <node pkg="nodelet" type="nodelet" name="image_proc_debayer_right"
            args="load image_proc/debayer stereo_nodelet_manager">
            <remap from="image_raw" to="right/image_raw" />
            <remap from="image_mono" to="right/image_mono" />
            <remap from="image_color" to="right/image_color" />
        </node>
    </group>

</launch>
